timer.hpp

```diff
@@ -198,29 +199,59 @@ AliquotTimer<F>& AliquotTimer<F>::operator()()
     double       now        = start_time;
     double       previous   = start_time;
     double       minimum    = limit;
+    // Initialize the variables to avoid `may be used uninitialized` error.
+    double       maximum    = 0.0;
+    double       last_m     = 0.0;
+    double       last_s     = 0.0;
+    // Compute the mean and the standard deviation on the fly using the
+    // algorithm from TAoCP Vol 2, section 4.2.2.
     int j = 0;
     for(; now < expiry_min || j < 100 && now < expiry_max; ++j)
         {
         f_();
         previous = now;
         now = timer.inspect();
-        if(now - previous < minimum)
+        double const delta = now - previous;
+        if(j == 0)
             {
-            minimum = now - previous;
+            minimum =
+            maximum = delta;
+
+            last_m = delta;
+            last_s = 0;
+            }
+        else
+            {
+            if(delta < minimum)
+                {
+                minimum = delta;
+                }
+            if(delta > maximum)
+                {
+                maximum = delta;
+                }
+            double const oldM = last_m;
+            last_m += (delta - oldM) / (j + 1);
+            last_s += (delta - oldM) * (delta - last_m);
             }
         }
     timer.stop();
     unit_time_ = minimum / dbl_freq;
+    const double std_dev = j > 0 ? std::sqrt(last_s/j) : 0;
```

Comment:
```suggestion
    double const std_dev = j > 0 ? std::sqrt(last_s/j) : 0;
```
Didn't coding rules test complain about this? If you didn't run it, please do.